metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -322,54 +323,54 @@ def _make_time_spine_data_set(
         """
         time_spine_table_alias = self._next_unique_table_alias()
 
-        agg_time_dimension_specs = {instance.spec for instance in agg_time_dimension_instances}
+        queried_specs = {instance.spec for instance in agg_time_dimension_instances}
         specs_required_for_where_constraints = {
             spec
             for constraint in time_spine_where_constraints
             for spec in constraint.linkable_spec_set.time_dimension_specs
         }
-        required_time_spine_specs = sorted(  # sorted for consistency in snapshots
-            agg_time_dimension_specs.union(specs_required_for_where_constraints),
+        required_specs = sorted(  # sorted for consistency in snapshots
```

Comment:
TODO: debug this for date part / maybe custom grain too. Either sort by qualified name or just leave the order as it was sent to you